code/modules/reagents/chemistry/items.dm

```diff
@@ -162,3 +131,198 @@
 			out_message += "<b>Analysis:</b> [R.description]\n"
 	to_chat(user, "[out_message.Join()]</span>")
 	desc = "An electrode attached to a small circuit box that will display details of a solution. Can be toggled to provide a description of each of the reagents. The screen currently displays detected vol: [round(cont.volume, 0.01)] detected pH:[round(cont.reagents.ph, 0.1)]."
+
+/obj/item/burner
+	name = "Alcohol burner"
+	desc = "A small table size burner used for heating up beakers."
+	icon = 'icons/obj/chemical.dmi'
+	icon_state = "burner"
+	grind_results = list(/datum/reagent/consumable/ethanol = 5, /datum/reagent/silicon = 10)
+	item_flags = NOBLUDGEON
+	resistance_flags = FLAMMABLE
+	w_class = WEIGHT_CLASS_TINY
+	heat = 2000
+	///If the flame is lit - i.e. if we're processing and burning
+	var/lit = FALSE
+	///total reagent volume
+	var/max_volume = 50
+	///What the creation reagent is
+	var/reagent_type = /datum/reagent/consumable/ethanol
+
+/obj/item/burner/Initialize()
+	. = ..()
+	create_reagents(max_volume, TRANSPARENT)//We have our own refillable - since we want to heat and pour
+	reagents.add_reagent(reagent_type, 15)
```

Comment:
```suggestion
	create_reagents(max_volume, TRANSPARENT)//We have our own refillable - since we want to heat and pour
	if(reagent_type)
		reagents.add_reagent(reagent_type, 15)
```
I assume you might want to be able to spawn empty burners at some point.

Reply:
yes that would be ideal